### PR TITLE
Update Proxy SHA to latest with libc++ 8.

### PR DESCRIPTION
**Pull request by PiotrSikora**
Signed-off-by: Piotr Sikora <piotrsikora@google.com>

### CODE REVIEW

File: pilot/docker/Dockerfile.proxyv2 [hunk 1]
```diff
@@ -8,7 +8,7 @@ RUN curl -s http://apt.llvm.org/llvm-snapshot.gpg.key | apt-key add -
 RUN echo "deb http://apt.llvm.org/xenial/ llvm-toolchain-xenial-7 main" \
     >/etc/apt/sources.list.d/llvm.list
 RUN apt-get update && apt-get install --no-install-recommends -y \
-    libc++-7-dev libc++abi-7-dev
+    libc++-8-dev libc++abi-8-dev
```
Comment (howardjohn) [inline pilot/docker/Dockerfile.proxyv2]: why do we need this here but not on master?

Comment (PiotrSikora) [reply pilot/docker/Dockerfile.proxyv2]: release-1.3+ is statically linked against libc++, so we don't need libc++ installed system-wide. Older releases are using older versions of Bazel, which didn't allow to do that (at least not trivially, and we were working against the deadline for HTTP/2 CVEs), so we ended up linking it dynamically in those older releases... Temporary glitch.

**Comment by PiotrSikora** [comment]
I think some of tests on Prow might be broken until we get https://github.com/istio/test-infra/pull/1702.

**Comment by PiotrSikora** [comment]
@howardjohn should `istio/ci:go1.11-k8s1.10.4-helm2.7.2-minikube0.25` be rebuilt using new `Dockerfile`?

Also, it seems that `e2e-pilot-cloudfoundry-v1alpha3-v2` is running on the VM, how did it pass in https://github.com/istio/istio/pull/16389?

**Comment by PiotrSikora** [comment]
OK, it seems that everything works (other than `prow/istio-pilot-multicluster-e2e.sh`, but it was already broken), so this should be good to go.

cc @istio/release-managers-1-1 @duderino

**Pull request merged**